daemon/k8s_watcher.go

```diff
@@ -1702,9 +1702,11 @@ func (d *Daemon) updatePodHostIP(pod *types.Pod) (bool, error) {
 		return true, fmt.Errorf("no/invalid HostIP: %s", pod.StatusHostIP)
 	}
 
-	podIP := net.ParseIP(pod.StatusPodIP)
-	if podIP == nil {
-		return true, fmt.Errorf("no/invalid PodIP: %s", pod.StatusPodIP)
+	if pod.StatusPodIP != "" {
+		podIP := net.ParseIP(pod.StatusPodIP)
+		if podIP == nil {
+			return true, fmt.Errorf("no/invalid PodIP: %s", pod.StatusPodIP)
+		}
```

Comment:
Does this mean we call `ipcache.IdentityCache.Upsert()` with an IP of `""` now?

Reply:
This needs to be pulled into its own thread its not iptables related.

Now with a IP of "" we will report "no/invalid PodIP..." in the logs. I see lots of these in the logs for every pod using host networking. With above change they get added to the ipcache. Alternatively we could probably get away with

```
if pod.StatusPodIP == "" {
        return false, nil
}
```

because we should have the nodeIP already in the cache from the NodeUpdate path.

Reply:
Removed this chunk from this PR.